sdk/identity/azure-identity/Troubleshoot.md

```diff
@@ -0,0 +1,223 @@
+them by following the instructions at their configuration links below.
+
+Azure Service | Managed Identity Configuration
+--- | --- |
+[Azure Virtual Machines](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token) | [Configuration Instructions](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/qs-configure-portal-windows-vm)
+[Azure App Service](https://docs.microsoft.com/azure/app-service/overview-managed-identity?tabs=python) | [Configuration Instructions](https://docs.microsoft.com/azure/app-service/overview-managed-identity?tabs=python)
+[Azure Kubernetes Service](https://docs.microsoft.com/azure/aks/use-managed-identity) | [Configuration Instructions](https://docs.microsoft.com/azure/aks/use-managed-identity)
+[Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/msi-authorization) |  |
+[Azure Arc](https://docs.microsoft.com/azure/azure-arc/servers/managed-identity-authentication) | [Configuration Instructions](https://docs.microsoft.com/azure/azure-arc/servers/security-overview#using-a-managed-identity-with-arc-enabled-servers)
+[Azure Service Fabric](https://docs.microsoft.com/azure/service-fabric/concepts-managed-identity) | [Configuration Instructions](https://docs.microsoft.com/azure/service-fabric/configure-existing-cluster-enable-managed-identity-token-service)
+
+## Troubleshooting Visual Studio Code Authentication Issues
+
+### Credential Unavailable Error
+
+#### Failed To Read VS Code Credentials / Authenticate via Azure Tools plugin in VS Code
+
+THe `VS Code Credential` failed to read the credential details from the cache.
+
+The Visual Studio Code authentication is handled by an integration with the Azure Account extension.
+To use this form of authentication, ensure that you have installed the Azure Account extension,
+then use View > Command Palette to execute the Azure: Sign In command. This command opens a browser window and displays a page that allows you 
+to sign in to Azure. After you've completed the login process, you can close the browser as directed. Running your application
+(either in the debugger or anywhere on the development machine) will use the credential from your sign-in.
+
+If you already had the Azure Account extension installed and had logged in to your account. Then try logging out and logging in again, as
+that will re-populate the cache on the disk and potentially mitigate the error you're getting.
+
+#### Msal Interaction Required Error
+
+THe `VS Code Credential` was able to read the cached credentials from the cache but the cached token is likely expired.
+Log into the Azure Account extension by via View > Command Palette to execute the Azure: Sign In command in the VS Code IDE.
+
+#### ADFS Tenant Not Supported
+
+The ADFS Tenants are not supported via the Azure Account extension in VS Code currently.
+The supported clouds are:
+
+Azure Cloud | Cloud Authority Host
```

Comment:
I think we can delete this table. These are the well-known clouds, but not the only ones we support, and the table doesn't help because you can't work around this error by authenticating in a different cloud.